.github/workflows/ci.yaml

```diff
@@ -1,74 +1,77 @@
 name: ci
 
+permissions:
+  packages: write
+
 on:
   pull_request:
+  workflow_dispatch:
   push:
     branches:
       - master
   schedule:
-    # See https://crontab.guru/weekly
-    - cron: 0 0 * * 0
+    # See https://crontab.guru/monthly
+    - cron: 0 0 1 * *
 
 jobs:
-  pre-commit:
+  build:
     runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v1
-      - name: Set PY
-        run:
-          echo "PY=$(python -c 'import hashlib,
-          sys;print(hashlib.sha256(sys.version.encode()+sys.executable.encode()).hexdigest())')"
-          >> $GITHUB_ENV
-      - uses: actions/cache@v1
-        with:
-          path: ~/.cache/pre-commit
-          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
-      - uses: pre-commit/action@v1.0.1
-
-  build-test-push:
-    runs-on: ubuntu-latest
-    needs: pre-commit
     strategy:
       fail-fast: false
       matrix:
         # Test modern Odoo versions with latest Postgres version
         pg_version:
+          - "16"
           - "15"
-          - "14"
           - "13"
-          - "12"
```

Comment:
Why removing these versions?

Reply:
I don't know, it was like this in the original code provided by @oskardotglobal. Any reason for it?

Reply:
I didn't have any need for them personally. Literally just that

Reply:
OK, then better to keep compatibility while working if there's no other reason to remove them